File: src/basics/src/listener.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void the_listener(const std_msgs::String::ConstPtr& the_thing_heard){

	ROS_INFO("Heard what : [%s]", the_thing_heard->data.c_str());
}

int main(int argc, char **argv){

	ros::init(argc, argv, "listener");
	ros::NodeHandle my_node;
	ros::Subscriber my_listener = my_node.subscribe("the_talker", 1, the_listener);
	ros::spin();
	return 0;
}

File: src/basics/scripts/listener.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import String

def the_listener(the_talk):
	the_heard_talk = the_talk.data
	rospy.loginfo("Heard what : %s", the_heard_talk)

def the_controller(): # or the_looper/spinner
	rospy.init_node('my_node', anonymous=True)
	rospy.Subscriber("the_talker", String, the_listener)
	rospy.spin()

if __name__ == '__main__':
	the_controller()
